irohad/ametsuchi/impl/postgres_block_index.cpp

```diff
@@ -5,125 +5,128 @@
 
 #include "ametsuchi/impl/postgres_block_index.hpp"
 
-#include <unordered_set>
+#include <numeric>
 
-#include <boost/range/adaptor/filtered.hpp>
+#include <boost/format.hpp>
 #include <boost/range/adaptor/indexed.hpp>
-#include <boost/range/algorithm/for_each.hpp>
-#include <boost/range/numeric.hpp>
 #include "common/types.hpp"
 #include "common/visitor.hpp"
-#include "cryptography/ed25519_sha3_impl/internal/sha3_hash.hpp"
 #include "interfaces/commands/command.hpp"
 #include "interfaces/commands/transfer_asset.hpp"
+#include "interfaces/common_objects/types.hpp"
 #include "interfaces/iroha_internal/block.hpp"
 
-namespace iroha {
-  namespace ametsuchi {
+namespace {
+  // Return transfer asset if command contains it
+  boost::optional<const shared_model::interface::TransferAsset &>
+  getTransferAsset(const shared_model::interface::Command &cmd) noexcept {
+    using ReturnType =
+        boost::optional<const shared_model::interface::TransferAsset &>;
+    return iroha::visit_in_place(
```

Comment:
Using `boost::get` with pointer may result in simpler code here, and visitor include will not be required.

Reply:
It will just replace visit_in_place with an if statement. I doubt there will be much improvement